test/Microsoft.SqlTools.ServiceLayer.Test.Common/TestLogger.cs

```diff
@@ -97,13 +97,15 @@ public void WriteWithCallstack()
 
         public void Verify(TraceEventType eventType, string message, string callstackMessage, bool shouldVerifyCallstack = false, bool expectLogMessage = true)
         {
+            Logger.Flush();
+            // The Regex uses .* between the severity and the message to allow SMO to vary the content. 140 SMO has nothing there, 150 has a timestamp
             if (expectLogMessage)
             {
-                Assert.True(File.Exists(Logger.LogFileFullPath) && Regex.IsMatch(LogContents, $@"\b{eventType}:\s+\d+\s+:\s+{message}", RegexOptions.Compiled));
+                Assert.True(File.Exists(Logger.LogFileFullPath) && Regex.IsMatch(LogContents, $@"\b{eventType}:.*{message}", RegexOptions.Compiled));
```

Comment:
The previous version of code was verifying that a trace eventid (like an error code) which is superimposed and emitted by the toolsservice logger is present in the log entry. So before we relax this verification I would like to understand why this relaxation is needed. Maybe there is a bug in my original implementation that you have uncovered. More likely this error code is just not present for the smo traces and we need to make this chagne only for those tests.

Reply:
eventid comes right after eventType. So for some reason the above check was getting satisfied before while it is not in smo. This does not look related to emission of date in smo log entry or is it?

Reply:
I think it's not practical for these tests to be so tightly coupled to the trace format of external components like SMO. What if we want to put something else in the trace between the message and the event type in the future? Having to also come back and update this test for something that should have no functional impact seems like excessive overhead. I changed this regex simply to allow SMO to have whatever it wants between the type and the message. The event id is still there afaik.

Reply:
The idea is to have one kind of verification logic for tools service logging and the other for smo logging. That will give you what you need. My concern is that if something breaks emission of the eventid in the toolsservice code then I want that to be caught by the tests.

Reply:
so you do not have to do much perhaps
revert the chagne to TestLogger
and change LoggerGetValidTableDefinitionTest 
to not call test.Verify() and instead do your own verify method
later if we have more tests for smo logging then we can create a dedicated test object for it

Reply:
David and I had an offline discussion and I see his point on over verifying the format, especially since in this case the eventid's presence in toolservice logging is dictated by the api when the log entry is generated. What gets written to the actual log file is purely determined by the listener.